libunwind/include/__libunwind_config.h

```diff
@@ -158,8 +158,8 @@
 #  if __riscv_xlen == 64
 #    define _LIBUNWIND_TARGET_RISCV 1
 #    ifdef __CHERI_PURE_CAPABILITY__
-#      define _LIBUNWIND_CONTEXT_SIZE 98
-#      define _LIBUNWIND_CURSOR_SIZE 121
+#      define _LIBUNWIND_CONTEXT_SIZE 96
```

Comment:
Why does context shrink by two but cursor only by one? But there should be static_assert()s so it must be fine.

Reply:
Hm, I wonder if I mis-counted before

Reply:
I think it's probably because of padding.

Reply:
Hmm, so here is the "old" cursor layout:

```
(gdb) ptype /o libunwind::UnwindCursor<libunwind::LocalAddressSpace, libunwind::Registers_riscv>
/* offset    |  size */  type = class libunwind::UnwindCursor<libunwind::LocalAddressSpace, libunwind::Registers_riscv> 
    [with A = class libunwind::LocalAddressSpace, 
         R = class libunwind::Registers_riscv] 
        : public libunwind::AbstractUnwindCursor {
                         private:
/*   16      |    16 */    class A &_addressSpace;
/*   32      |   784 */    R _registers;
/*  816      |   144 */    unw_proc_info_t _info;
/*  960      |     1 */    bool _unwindInfoMissing;
/*  961      |     1 */    bool _isSignalFrame;
/* XXX 14-byte padding */

                           /* total size (bytes):  976 */
                         }
```

That should result in 976 / 8 == 122 for the cursor size.  I'm not sure why this is not tripping up the assertion in the current code.  The new size is 960 bytes which results in the 120 in this commit.

Humm, the existing RISC-V world builds with this set to 122 as well as 121.  Given how picky the check_fit<> template has been in the past, I don't know why.

Oh:

```
struct __attribute__((aligned(sizeof(void*)))) unw_cursor_t {
  uint64_t data[_LIBUNWIND_CURSOR_SIZE];
};
```

That alignment means that too-small values can work since it gets padded out.  This is the existing unw_cursor_t:

```
(gdb) ptype /o unw_cursor_t 
/* offset    |  size */  type = struct unw_cursor_t {
/*    0      |   968 */    uint64_t data[121];
/* XXX  8-byte padding */

                           /* total size (bytes):  976 */
                         }
```

Reply:
This is probably also my fault as I probably just used the smallest value which satisfied the assertion failure when doing RISC-V the first time.

Reply:
I feel like I was the one responsible for this code. I probably did something stupid like work out how many pointers there were in the struct and just add that on again, forgetting about padding for alignment.